### PR TITLE
[TEST] [DELTA] Refine CI and add test detla 1.0.0 with Spark 3.1

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

- Fix `-Pkyuubi-sql-spark_3.1` introduced by KYUUBI #631 in GH action
- Tune GH action and travis scripts style
- Enable tests of detla 1.0.0 against Spark 3.1

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1104,6 +1104,12 @@
                 <groupId>io.delta</groupId>
                 <artifactId>delta-core_${scala.binary.version}</artifactId>
                 <version>${delta.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
```
Comment (pan3793) [inline pom.xml:1112]: https://github.com/delta-io/delta/pull/676

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -34,11 +34,11 @@ jobs:
...
-          - '-Pspark-3.1 -Pkyuubi-sql-spark_3.1'
-          - '-Pspark-3.1 -Pkyuubi-sql-spark_3.1 -Pspark-hadoop-3.2'
-          - '-Pspark-3.2-snapshot -pl :kyuubi-spark-sql-engine,:kyuubi-common,:kyuubi-ha,:kyuubi-zookeeper -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DataLakeTest'
-          - '-DwildcardSuites=org.apache.kyuubi.operation.tpcds.TPCDSOutputSchemaSuite,org.apache.kyuubi.operation.tpcds.TPCDSDDLSuite -Dmaven.plugin.scalatest.exclude.tags=""'
+          - '-Pspark-3.0 -Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.1.1 -Dspark.archive.name=spark-3.1.1-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest'
+          - '-Pspark-3.1 -Pkyuubi-extension-spark_3.1'
+          - '-Pspark-3.1 -Pkyuubi-extension-spark_3.1 -Pspark-hadoop-3.2'
+          - '-Pspark-3.2-snapshot -pl :kyuubi-spark-sql-engine,:kyuubi-common,:kyuubi-ha,:kyuubi-zookeeper'
+          - '-Pspark-3.0 -Dtpcds.tests=true'
```
Comment (pan3793) [inline .github/workflows/master.yml:41]: `-Dtpcds.tests=true` affect nothing, just for following steps matching

**Comment by pan3793** [comment]
Travis failed with 
> The job exceeded the maximum log length, and has been terminated.

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -34,11 +34,11 @@ jobs:
...
-          - '-Pspark-3.1 -Pkyuubi-sql-spark_3.1'
-          - '-Pspark-3.1 -Pkyuubi-sql-spark_3.1 -Pspark-hadoop-3.2'
-          - '-Pspark-3.2-snapshot -pl :kyuubi-spark-sql-engine,:kyuubi-common,:kyuubi-ha,:kyuubi-zookeeper -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DataLakeTest'
-          - '-DwildcardSuites=org.apache.kyuubi.operation.tpcds.TPCDSOutputSchemaSuite,org.apache.kyuubi.operation.tpcds.TPCDSDDLSuite -Dmaven.plugin.scalatest.exclude.tags=""'
+          - '-Pspark-3.0 -Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.1.1 -Dspark.archive.name=spark-3.1.1-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest'
+          - '-Pspark-3.1 -Pkyuubi-extension-spark_3.1'
+          - '-Pspark-3.1 -Pkyuubi-extension-spark_3.1 -Pspark-hadoop-3.2'
```
Comment (ulysses-you) [inline .github/workflows/master.yml:39]: ah, seems we need to backport into branch-1.2.0

File: kyuubi-common/src/test/java/org/apache/kyuubi/tags/DeltaTest.java [hunk 4]
```diff
@@ -27,4 +27,4 @@
 @TagAnnotation
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.METHOD, ElementType.TYPE})
-public @interface DataLakeTest {}
+public @interface DeltaTest {}
```
Comment (ulysses-you) [inline kyuubi-common/src/test/java/org/apache/kyuubi/tags/DeltaTest.java:30]: Not a big issue, but shall we add a parent tag for `DeltaTest` and `IcebergTest` ?

File: pom.xml (hunk 1 above)
Comment (ulysses-you) [reply pom.xml:1112]: It's better to add some comments at this code place or add a `todo` to remind update this in future ?

**Review by ulysses-you — COMMENTED** [review]
thank you @pan3793 for driving this.

File: .github/workflows/master.yml (hunk 3 above)
Comment (pan3793) [reply .github/workflows/master.yml:39]: yes

File: kyuubi-common/src/test/java/org/apache/kyuubi/tags/DeltaTest.java (hunk 4 above)
Comment (pan3793) [reply kyuubi-common/src/test/java/org/apache/kyuubi/tags/DeltaTest.java:30]: I think we use tags here to mark tests which not support specific Spark versions, and due to the limitation of scalatest maven plugin, such as collection operation, the workflow scripts are not elegant now. Would you explain the benefit of adding a parent tag?

File: pom.xml (hunk 1 above)
Comment (pan3793) [reply pom.xml:1120]: updated

File: kyuubi-common/src/test/java/org/apache/kyuubi/tags/DeltaTest.java (hunk 4 above)
Comment (ulysses-you) [reply kyuubi-common/src/test/java/org/apache/kyuubi/tags/DeltaTest.java:30]: Not a strong opinion. I'm fine with both.

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
mistouch the close button : (

File: .github/workflows/master.yml [hunk 5]
```diff
@@ -72,16 +72,18 @@ jobs:
...
+        if: ${{ contains(matrix.profiles, 'tpcds.tests=true') }}
+        run: |
+          mvn clean install ${{ matrix.profiles }} -DskipTests -pl :kyuubi-spark-sql-engine,:kyuubi-common,:kyuubi-ha,:kyuubi-zookeeper
+          mvn test ${{ matrix.profiles }} -Dtest=none -DwildcardSuites=org.apache.kyuubi.operation.tpcds.TPCDSOutputSchemaSuite,org.apache.kyuubi.operation.tpcds.TPCDSDDLSuite
       - name: Code coverage
         if: ${{ matrix.profiles == '' }}
         run: bash <(curl -s https://codecov.io/bash)
       - name: Detected Dependency List Change
-        if: ${{ ! contains(matrix.profiles, 'spark-3.2-snapshot') && ! contains(matrix.profiles, 'wildcardSuites') }}
+        if: ${{ ! contains(matrix.profiles, 'spark-3.2-snapshot') && ! contains(matrix.profiles, 'tpcds.tests=true') }}
```
Comment (yaooqinn) [inline .github/workflows/master.yml:86]: instead of using matrix I prefer to move tpcds tests to another separate job like `rat`

Comment (yaooqinn) [reply .github/workflows/master.yml:86]: And not related to this PR, we can mv `spark-3.2-snapshot` to a another `yml` with nightly build enabled?

File: pom.xml [hunk 6]
```diff
@@ -1550,20 +1564,27 @@
...
-                <spark.version>3.2.0-SNAPSHOT</spark.version>
+                <spark.version>3.1.1</spark.version>
+                <delta.version>1.0.0</delta.version>
+                <maven.plugin.scalatest.exclude.tags>org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.IcebergTest</maven.plugin.scalatest.exclude.tags>
             </properties>
         </profile>
 
         <profile>
-            <id>spark-3.1</id>
+            <id>spark-3.2-snapshot</id>
```
Comment (yaooqinn) [inline pom.xml:1584]: spark-master？

Comment (pan3793) [reply pom.xml:1584]: just reordered `spark-3.1` and `spark-3.2-snapshot`

Comment (pan3793) [reply pom.xml:1584]: sounds good

File: .travis.yml [hunk 7]
```diff
@@ -31,36 +31,45 @@ matrix:
...
+      mvn clean install $PROFILE -Dmaven.javadoc.skip=true -V
+    fi
+  - |
+    if [[ "$JOB" == "nightly-build" ]]; then
+      mvn clean install $PROFILE -pl :kyuubi-spark-sql-engine,:kyuubi-common,:kyuubi-ha,:kyuubi-zookeeper -Dmaven.javadoc.skip=true -V
+    fi
+  - |
+    if [[ "$JOB" == "tpcds-tests" ]]; then
+      mvn clean install $PROFILE -DskipTests -pl :kyuubi-spark-sql-engine,:kyuubi-common,:kyuubi-ha,:kyuubi-zookeeper
+      mvn test $PROFILE -Dtest=none -DwildcardSuites=org.apache.kyuubi.operation.tpcds.TPCDSOutputSchemaSuite,org.apache.kyuubi.operation.tpcds.TPCDSDDLSuite
```
Comment (yaooqinn) [inline .travis.yml:68]: org.apache.kyuubi.operation.tpcds.TPCDSOutputSchemaSuite,org.apache.kyuubi.operation.tpcds.TPCDSDDLSuite

can we rename them to xxxTPCDSSuite then `-DwildcardSuites=*TPCDSSuite` for short?

File: .github/workflows/master.yml [hunk 8]
```diff
@@ -91,3 +85,43 @@ jobs:
           path: |
             **/target/unit-tests.log
             **/kyuubi-spark-sql-engine.log*
+
+  tpcds:
+    name: TPC-DS Schema Tests
```
Comment (yaooqinn) [inline .github/workflows/master.yml:90]: not only schema? simply `TPC-DS Tests`?

File: .github/workflows/master.yml [hunk 9]
```diff
@@ -91,3 +85,43 @@ jobs:
...
+          key: ${{ runner.os }}-maven-net-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-net-
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository/io
+          key: ${{ runner.os }}-maven-io-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-io-
+      - name: Run TPC-DS Schema Tests
```
Comment (yaooqinn) [inline .github/workflows/master.yml:124]: ditto

File: .travis.yml [hunk 10]
```diff
@@ -31,36 +31,45 @@ matrix:
...
-        - EXCLUDE_TAGS="org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DataLakeTest"
+        - PROFILE="-Pspark-3.0 -Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.1.1 -Dspark.archive.name=spark-3.1.1-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest"
+        - JOB="build"
     - name: Test Kyuubi w/ -Pspark-3.1 -Pkyuubi-extension-spark_3.1
       env:
         - PROFILE="-Pspark-3.1 -Pkyuubi-extension-spark_3.1"
-        - EXCLUDE_TAGS="org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DataLakeTest"
-    - name: Test Kyuubi w/ -Pspark-3.1 -Pspark-hadoop-3.2 -Pkyuubi-extension-spark_3.1
+        - JOB="build"
+    - name: Test Kyuubi w/ -Pspark-master w/ Spark 3.2 nightly build
```
Comment (yaooqinn) [inline .travis.yml:44]: remove `w/ Spark 3.2 nightly build`？

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks for review, got it merged for v1.3.0, and will send following PRs for 1. nightly build; 2. TPC-DS test against Delta

**Pull request merged**